app/src/lib/infer-last-push-for-repository.ts

```diff
@@ -46,7 +46,7 @@ export async function inferLastPushForRepository(
     }
   }
 
-  if (repository.gitHubRepository !== null) {
+  if (lastPushDate === null && repository.gitHubRepository !== null) {
```

Comment:
This change makes little sense when viewed like this but if you expand the diff upwards you'll see that this is a local variable that's set when we've fetched the repository associated with the current remote. In other words not having this check is what's been causing us to do two requests for the same repository in rapid succession when refreshing indicators

![image](https://user-images.githubusercontent.com/634063/91544418-03312880-e920-11ea-83b1-964e1be1f63e.png)

It doesn't have any measurable effect on performance since the previous request will be cached and but it's nevertheless redundant and fills our network log with unnecessary requests.

Reply:
Worked around this using early returns instead